### PR TITLE
ref: Remove demangling from C/Python bindings

### PR DESCRIPTION
**Pull request by loewenheim**
As discussed under https://github.com/getsentry/symbolic/pull/699.

TODO: The example in the readme involves demangling, what should we put there instead?

### CODE REVIEW

File: py/symbolic/symcache.py [hunk 1]
```diff
@@ -38,11 +37,6 @@ def __init__(self, sym_addr, instr_addr, line, lang, symbol, full_path=None):
...
         self.full_path = full_path or None
 
-    @property
-    def function_name(self):
-        """The demangled function name."""
-        return demangle_name(self.symbol, lang=self.lang)
-
     def __str__(self):
         return "%s:%s (%s)" % (
             self.function_name,
```
Comment (Swatinem) [inline py/symbolic/symcache.py:42]:
```suggestion
            self.symbol,
```

**Review by Swatinem — APPROVED** [review]
good call, no idea though.

I would prefer if we stopped "advertising" cabi/python in the first place. Its a pain to maintain and people can easily generate their own bindings via bindgen or pyo3 or whatever the best approach nowadays is :-)

**Comment by loewenheim** [comment]
Would you object to just removing the example?

**Comment by Swatinem** [comment]
> Would you object to just removing the example?

I think thats a fair thing to do.

**Review by Swatinem — APPROVED** [review]
pls make sure to call this out as a breaking change in the changelog

**Pull request merged**